tests/recognizer-tests.js

```diff
@@ -30,6 +30,16 @@ test("A unicode route recognizes", function() {
   equal(router.recognize("/uniçø∂∑"), null);
 });
 
+test("A unicode with special char route recognizes", function() {
```

Comment:
This isn't a unicode test, this is RFC 3986 special cases. Write tests for all of them: "*, !:"